implementations/micrometer-registry-timestream2/src/main/java/io/micrometer/timestream2/TimestreamMeterRegistry.java

```diff
@@ -0,0 +1,333 @@
+        }
+
+        // VisibleForTesting
+        Stream<Record> metricData(Meter m) {
+            return stream(m.measure().spliterator(), false)
+                    .map(ms -> record(m.getId().withTag(ms.getStatistic()),
+                            null,
+                            MeasureValueType.DOUBLE, //default type ?
+                            ms.getValue()))
+                    .filter(Objects::nonNull);
+        }
+
+        @Nullable
+        private Record record(Meter.Id id, @Nullable String suffix, MeasureValueType measureValueType, double value) {
+            if (Double.isNaN(value)) {
+                return null;
+            }
+            List<Tag> tags = id.getConventionTags(config().namingConvention());
+            return Record.builder()
+                    .measureName(getMetricName(id, suffix))
+                    .measureValue(String.valueOf(clampMetricValue(value)))
+                    .measureValueType(measureValueType)
+                    .dimensions(toDimensions(tags))
+                    .time(String.valueOf(timestamp.getEpochSecond()))
+                    .timeUnit(SECONDS)
+                    .build();
+        }
+
+        // VisibleForTesting
+        String getMetricName(Meter.Id id, @Nullable String suffix) {
+            String name = suffix != null ? id.getName() + "." + suffix : id.getName();
+            return config().namingConvention().name(name, id.getType(), id.getBaseUnit());
+        }
+
+        private List<Dimension> toDimensions(List<Tag> tags) {
+            return tags.stream()
+                    .filter(this::isAcceptableTag)
+                    .map(tag -> Dimension.builder()
+                            .name(tag.getKey())
+                            .value(tag.getValue())
+                            .dimensionValueType(DimensionValueType.VARCHAR)
+                            .build()
+                    )
+                    .collect(toList());
+        }
+
+        private boolean isAcceptableTag(Tag tag) {
+            if (StringUtils.isBlank(tag.getValue())) {
+                warnThenDebugLogger.log("Dropping a tag with key '" + tag.getKey() + "' because its value is blank.");
+                return false;
+            }
+            return true;
+        }
+
+        /**
+         * Clean up metric to be within the allowable range}
+         *
+         * @param value unsanitized value
+         * @return value clamped to allowable range
+         */
+        double clampMetricValue(double value) {
+            // Leave as is and let the SDK reject it
+            if (Double.isNaN(value)) {
+                return value;
+            }
+            double magnitude = Math.abs(value);
+            if (magnitude == 0) {
+                // Leave zero as zero
+                return 0;
+            }
+            // Non-zero magnitude, clamp to allowed range
+            double clampedMag = Math.min(magnitude, Double.MAX_VALUE);
```

Comment:
I have a few questions here:

1. Does AWS let you pass any `double`?
2. If so and any `double` is allowed, is this whole method really needed?
3. What about the smallest positive nonzero value, is there any?

Or is this for handling negative/positive infinity?

Reply:
1. `DOUBLE` is a `64-bit variable-precision implementing the IEEE Standard 754 for Binary Floating-Point Arithmetic.`
2. I took this method from `io.micrometer.cloudwatch2.CloudWatchMeterRegistry` but I did not check if  NaN, +Infinity, -Infinity are supported on Timestream.